diff --git a/group.module b/group.module
index fe030ad..b135062 100644
--- a/group.module
+++ b/group.module
@@ -498,6 +498,7 @@ function group_content_wizard_cancel($form, FormStateInterface $form_state) {
 function group_content_entity_submit($form, FormStateInterface $form_state) {
   /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
   $entity = $form_state->getFormObject()->getEntity();
+  $entityType = $entity->getEntityType()->id();
 
   /** @var \Drupal\group\Entity\GroupInterface $group */
   $group = $form_state->get('group');
@@ -507,14 +508,16 @@ function group_content_entity_submit($form, FormStateInterface $form_state) {
   // without using the 2-step wizard. We can therefore safely assume the user
   // wants to see the entity itself and not the relationship (group content).
   // This only applies if there was no destination GET parameter set in the URL.
-  if ($entity->access('view')) {
-    $form_state->setRedirectUrl($entity->toUrl());
-  }
-  elseif ($group->access('view')) {
-    $form_state->setRedirectUrl($group->toUrl());
-  }
-  else {
-    $form_state->setRedirect('<front>');
+  if ($entityType != 'taxonomy_term') {
+    if ($entity->access('view')) {
+      $form_state->setRedirectUrl($entity->toUrl());
+    }
+    elseif ($group->access('view')) {
+      $form_state->setRedirectUrl($group->toUrl());
+    }
+    else {
+      $form_state->setRedirect('<front>');
+    }
   }
 }
 
